dartagnan/src/main/java/com/dat3m/dartagnan/program/processing/ProcessingManager.java

```diff
@@ -74,19 +74,26 @@ public class ProcessingManager implements ProgramProcessor {
 
     private ProcessingManager(Configuration config) throws InvalidConfigurationException {
         config.inject(this);
-
+        final Intrinsics intrinsics = Intrinsics.newInstance();
+        final FunctionProcessor sccp = constantPropagation ? SparseConditionalConstantPropagation.fromConfig(config) : null;
         programProcessors.addAll(Arrays.asList(
-                printBeforeProcessing ? DebugPrint.withHeader("Before processing", Printer.Mode.ALL) : null,
+                intrinsics.markIntrinsicsPass(),
+                GEPToAddition.newInstance(),
+                RegisterDecomposition.newInstance(),
                 StaticMemoryInitializer.newInstance(),
+                printBeforeProcessing ? DebugPrint.withHeader("Before processing", Printer.Mode.ALL) : null,
```

Comment:
shouldn't this be the very first pass?

Reply:
Probably yes (minus potentially the `StaticMemoryInitializer`, but that one is only there for Boogie code).